server/consumer.go

```diff
@@ -1718,7 +1729,14 @@ func (o *consumer) processNextMsgReq(_ *subscription, c *client, _, reply string
 			break
 		}
 	}
+	sendq := o.sendq
 	o.mu.Unlock()
+
+	if shouldRespond {
+		// Empty message meaning that there were no errors.
+		pmsg := &jsPubMsg{reply, reply, _EMPTY_, nil, nil, nil, 0}
```

Comment:
In case the `sub.Pull()` is done from another account, then this would not deliver the response.

Reply:
This should work ok once I push the current branch. Sync with that should work fine.

Reply:
This should not be needed. Let's chat in am.

Reply:
Just tried with latest and seems that for the approach in this PR, it still needs to transform the reply inbox. An alternative that I was thinking to make the sub.Poll() synchronous is to handle as an old style request on the client side: https://github.com/nats-io/nats.go/blob/118341e3023b7bf5bf01b3f332b98fc52e2c22b1/js.go#L913-L980